javaparser-core/src/main/java/com/github/javaparser/ParserConfiguration.java

```diff
@@ -124,4 +125,18 @@ public ParserConfiguration setLexicalPreservationEnabled(boolean lexicalPreserva
     public boolean isLexicalPreservationEnabled() {
         return lexicalPreservationEnabled;
     }
+
+    /**
+     * Retrieve the SymbolResolver to be used while parsing, if any.
+     */
+    public SymbolResolver getSymbolResolver() {
+        return symbolResolver;
```

Comment:
No nulls please, `Optional.ofNullable` to the rescue :-)

Reply:
Background: not a single method in JavaParser returns null anymore since 3.0.0. Mixing `Optional` and null will break the rule, and people would again have to null-check everything.

Reply:
Right, doing